library/lib.sh

```diff
@@ -373,13 +373,15 @@ lsrUploadLogs() {
 }
 
 lsrArrtoStr() {
+    local keys values key value
+    local arr_name=$1
     # Convert associative array into a space separated key=value list
-    eval "arr=(\${!$1[@]})"
-    # arr and val are defined above with eval
-    # shellcheck disable=SC2154
-    for k in "${arr[@]}"; do
-        eval "val=\${$1[$k]}"
-        printf "%s=%s " "$k" "$val"
+    eval "keys=(\${!$arr_name[@]})"
+    eval "values=(\${$arr_name[@]})"
+    for i in $(seq 0 $((${#keys[@]} - 1))); do
+        key="${keys[$i-1]}"
```

Comment:
If `i` is `0`, then this will do `key="${keys[-1]}"` ?

Reply:
Forgot to fix it here, fixed now

Reply:
still using `keys[$i-1]`?

Reply:
Also - does is it guaranteed that the order of the keys is exactly the same as the order of the values?

Reply:
After thinking about this some more - it would be better to have `ANSIBLE_ENVS` to be a simple array, not an associative array.

Setting an env var would be like this:
```
ANSIBLE_ENVS+=("ANSIBLE_GATHERING=$value")
```
then you could use it on the command line like this:
```
        cmd="${ANSIBLE_ENVS[*]} ansible-playbook -i $inventory $skip_tags $limit process_python_modules_packages.yml -vv"
        rlRun "$cmd -e packages_file=$packages -e logfile=$LOGFILE &> $LOGFILE.modules" 0 "process python modules"
```
no function needed - this assumes none of the `$value` have spaces or shell metacharacters